lib/sqlalchemy/ext/asyncio/scoping.py

```diff
@@ -67,7 +67,7 @@
     AsyncSession,
     ":class:`_asyncio.AsyncSession`",
     ":class:`_asyncio.scoping.async_scoped_session`",
-    classmethods=["close_all", "object_session", "identity_key"],
+    classmethods=["object_session", "identity_key"],
```

Comment:
**Federico Caselli** (CaselIT) wrote:

I think think should work ok if we keep this as generated by the script

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4878

Reply:
As I mentioned in https://github.com/sqlalchemy/sqlalchemy/pull/10429#discussion_r1349477020, script will generate the following code, which seems weird (though it works) and has wrong type annotation. Should I accept it?
```diff
--- /workspaces/sqlalchemy/.tox/lint/lib/python3.10/site-packages/sqlalchemy/ext/asyncio/scoping.py
+++ <proposed changes>
@@ -1544,6 +1544,21 @@
         return self._proxied.info
 
     @classmethod
+    def close_all(cls) -> None:
+        r"""Close all :class:`_asyncio.AsyncSession` sessions.
+
+        .. container:: class_bases
+
+            Proxied for the :class:`_asyncio.AsyncSession` class on
+            behalf of the :class:`_asyncio.scoping.async_scoped_session` class.
+
+        .. deprecated:: 2.0 The :meth:`.AsyncSession.close_all` method is deprecated and will be removed in a future release.  Please refer to :func:`.session.close_all_sessions`.
+
+        """  # noqa: E501
+
+        return AsyncSession.close_all()
+
+    @classmethod
     def object_session(cls, instance: object) -> Optional[Session]:
         r"""Return the :class:`.Session` to which an object belongs.
```

Reply:
Oh, I find out that `@util.deprecated` will effect `inspect.iscoroutinefunction` in generate_proxy_methods.py, which makes it think `AsyncSession.close_all()` is not an async method. [`inspect.markcoroutinefunction()`](https://docs.python.org/3.12/library/inspect.html#inspect.markcoroutinefunction) in Python 3.12 may help though, we may have to use some workaround now.

Reply:
the bug in on @util.deprecated then. I can take a look if you like

Reply:
Please take a look for me. I have no idea how to fix it.

Reply:
**mike bayer** (zzzeek) wrote:

Done

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4878